src/main/java/emissary/command/WhatCommand.java

```diff
@@ -174,29 +173,22 @@ protected void processFilePath(final Path path) throws ParserException, IOExcept
         }
 
         // Using Header
-        RandomAccessFile raf = null;
-        try {
-            final SessionParser parser = parserFactory.makeSessionParser(Files.newByteChannel(path));
-            LOG.debug("Using parser " + parser.getClass().getName() + " for " + path);
-            final SessionProducer producer = new SessionProducer(parser, Form.UNKNOWN);
-            final String parent = path.getParent().toString();
-            final String fileName = path.getFileName().toString();
-            while (true) {
-                try {
-                    final DecomposedSession sessionHash = parser.getNextSession();
-                    final IBaseDataObject payload = producer.createAndLoadDataObject(sessionHash, parent);
-                    final Identification id = identify(payload, this.headerIdentification);
-
-                    LOG.info(payload.getFilename() + "/" + fileName + ": " + id.getTypeString() + " [" + payload.dataLength() + "]");
-                } catch (ParserEOFException eof) {
-                    LOG.debug("Parser reached end of file: ", eof);
-                    // Expected EOF
-                    break;
-                }
-            }
-        } finally {
-            if (raf != null) {
-                raf.close();
+        final SessionParser parser = parserFactory.makeSessionParser(Files.newByteChannel(path));
+        LOG.debug("Using parser " + parser.getClass().getName() + " for " + path);
+        final SessionProducer producer = new SessionProducer(parser, Form.UNKNOWN);
+        final String parent = path.getParent().toString();
+        final String fileName = path.getFileName().toString();
+        while (true) {
+            try {
+                final DecomposedSession sessionHash = parser.getNextSession();
+                final IBaseDataObject payload = producer.createAndLoadDataObject(sessionHash, parent);
+                final Identification id = identify(payload, this.headerIdentification);
+
+                LOG.info(payload.getFilename() + "/" + fileName + ": " + id.getTypeString() + " [" + payload.dataLength() + "]");
+            } catch (ParserEOFException eof) {
+                LOG.debug("Parser reached end of file: ", eof);
+                // Expected EOF
+                break;
```

Comment:
Sonarlint doesn't like this loop. It could be changed to either evaluate `parser.isFullyParsed()`, or update some boolean when eof is reached.

Reply:
I am just trying to get rid of `raf` and fix the bug for code not reachable. Changing the while loop is out of scope for this pr.